doc: Adds documentation and example for mongodbatlas_flex_cluster resource and data source (#2744)
mongodb/terraform-provider-mongodbatlas: docs/resources/flex_cluster.md

}

output "mongodbatlas_flex_cluster" {
  value = data.mongodbatlas_flex_cluster.flex_cluster.name
}

output "mongodbatlas_flex_clusters_names" {
  value = [for cluster in data.mongodbatlas_flex_clusters.flex_clusters.results : cluster.name]
}
```

<!-- schema generated by tfplugindocs -->
## Schema

### Required

- `name` (String) Human-readable label that identifies the instance.
- `project_id` (String) Unique 24-hexadecimal character string that identifies the project.
- `provider_settings` (Attributes) Group of cloud provider settings that configure the provisioned MongoDB flex cluster. (see [below for nested schema](#nestedatt--provider_settings))

### Optional

- `tags` (Map of String) Map that contains key-value pairs between 1 to 255 characters in length for tagging and categorizing the instance.
- `termination_protection_enabled` (Boolean) Flag that indicates whether termination protection is enabled on the cluster. If set to `true`, MongoDB Cloud won't delete the cluster. If set to `false`, MongoDB Cloud will delete the cluster.

### Read-Only

- `backup_settings` (Attributes) Flex backup configuration (see [below for nested schema](#nestedatt--backup_settings))

> - `cluster_type` (String) Flex cluster topology.

Reviewer: Do we have any more details about `cluster_type`?

Reviewer 2: This is what was generated from the [API](https://github.com/search?q=repo%3A10gen%2Fmms+%22Flex+cluster+topology%22&type=code).

Read-Only:

- `disk_size_gb` (Number) Storage capacity available to the flex cluster expressed in gigabytes.
- `provider_name` (String) Human-readable label that identifies the cloud service provider.

<a id="nestedatt--backup_settings"></a>
### Nested Schema for `backup_settings`

Read-Only:

- `enabled` (Boolean) Flag that indicates whether backups are performed for this flex cluster. Backup uses [TODO](TODO) for flex clusters.

<a id="nestedatt--connection_strings"></a>
### Nested Schema for `connection_strings`

Read-Only:

- `standard` (String) Public connection string that you can use to connect to this cluster. This connection string uses the mongodb:// protocol.
- `standard_srv` (String) Public connection string that you can use to connect to this flex cluster. This connection string uses the `mongodb+srv://` protocol.

# Import 

> Flex Cluster resource can be imported using the Project ID and Flex Cluster name, in the format `PROJECT_ID-FLEX_CLUSTER_NAME`, e.g.

Reviewer 3: ```suggestion
You can import the Flex Cluster resource by using the Project ID and Flex Cluster name, in the format `PROJECT_ID-FLEX_CLUSTER_NAME`. For example:
```
